Extensions/one_click_postage.js

```diff
@@ -568,130 +455,105 @@ XKit.extensions.one_click_postage = new Object({
 		XKit.tools.init_css("one_click_postage");
 
 		// Let's first check if we have auto_tagger installed and active.
-		XKit.extensions.one_click_postage.get_autotagger();
+		this.get_autotagger();
 
 		// Then, check Quick Tags.
-		XKit.extensions.one_click_postage.get_quicktags();
+		this.get_quicktags();
 
 		if (this.preferences.allow_resize.value) {
 			XKit.tools.add_css("#x1cpostage_caption { resize: vertical; }", "one_click_postage_resize");
 		}
 
 		if (this.preferences.show_small_ui.value) {
 			this.caption_height = 50;
-			var slim_css = "#x1cpostage_caption { height: 50px; }" +
-					"#x1cpostage_reblog, #x1cpostage_queue, #x1cpostage_draft { height: 32px; }" +
-					"#x1cpostage_reblog i::before, #x1cpostage_queue i::before, #x1cpostage_draft i::before { position: static; }";
-			XKit.tools.add_css(slim_css, "one_click_postage_slim");
+			XKit.tools.add_css(
+				"#x1cpostage_caption { height: 50px; }" +
+				"#x1cpostage_reblog, #x1cpostage_queue, #x1cpostage_draft { height: 32px; }" +
+				"#x1cpostage_reblog i::before, #x1cpostage_queue i::before, #x1cpostage_draft i::before { position: static; }",
+			"one_click_postage_slim");
+		}
+
+		if (this.preferences.show_reverse_ui.value) {
+			XKit.tools.add_css(
+				"#x1cpostage_draft { border-radius: 0px 0px 3px 0px; }" +
+				"#x1cpostage_reblog { border-radius: 0px 0px 0px 3px; }" +
+				"#x1cpostage_tags { border-radius: 3px 3px 0px 0px; border-bottom: 0; }" +
+				"#x1cpostage_replace { border-bottom: 0; }" +
+				"#x1cpostage_remove_caption { border-top: 1px solid #abafbc; border-bottom: 0; }",
+			"x1cpostage_reverse_ui");
 		}
 
 		XKit.blog_listener.add("one_click_postage", this.init);
 	},
 
-	init: function(m_blogs) {
+	init: function(blogs) {
 
-		var m_remove_button = "<div id=\"x1cpostage_remove_caption\">remove caption</div>";
+		this.default_blog_id = this.preferences.default_blog.value || blogs[0];
+		this.blogs_list = blogs;
 
-		if (!this.preferences.show_caption_remover.value) {
-			m_remove_button = "";
+		var blog_html = "";
+		for (var x in blogs) {
+			blog_html += '<option value="' + blogs[x] + '"' + (blogs[x] == this.default_blog_id ? ' selected' : '') + '>' + blogs[x] + '</option>';
 		}
 
-		var m_remove_box_style = "";
-		if (!this.preferences.show_caption.value) {
-			m_remove_box_style = " style=\"display: none;\" ";
-		}
+		var ui_html = [
+			'<div id="x1cpostage_reblog"><i>&nbsp;</i></div>' + '<div id="x1cpostage_queue"><i>&nbsp;</i></div>' + '<div id="x1cpostage_draft"><i>&nbsp;</i></div>',
 
-		var m_clear_tags_button = "<div id=\"x1cpostage_clear_tags\">clear tags</div>";
+			(this.preferences.show_social.value ?
```

Comment:
```
if (this.preferences.show_social.value) {
  ui_html.append(whatever)
}
```
is probably more readable then this repeated `conditional ? thing : ''` pattern in an array literal